src/main/java/com/datasift/client/DataSiftIngestionResult.java

```diff
@@ -0,0 +1,119 @@
+package com.datasift.client;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.List;
+
+public class DataSiftIngestionResult implements DataSiftResult, IngestionRateLimit {
+    @JsonProperty
+    protected String error;
+    protected Response response;
+    private Throwable cause;
+    private boolean failed;
+
+    @Override
+    public boolean isSuccessful() {
+        return !failed
+                && response != null
+                && error == null
+                && response.status() > 199
+                && response.status() < 400;
+    }
+
+    @Override
+    public Response getResponse() {
+        return response;
+    }
+
+    @Override
+    public void setResponse(Response response) {
+        this.response = response;
+        //setResponse can happen after a result has been marked as failed
+        //so if it is already failed then keep it as a failure
+        failed = failed ? failed : response.hasFailed();
+        //likewise if already failed keep the existing cause
+        cause = cause != null ? cause : response.failureCause();
+    }
+
+    @Override
+    public Throwable failureCause() {
+        return cause;
+    }
+
+    @Override
+    public boolean isAuthorizationSuccesful() {
+        return isSuccessful() && response.status() != 401;
+    }
+
+    @Override
+    public int requestRateLimit() {
+        List<String> limit = response.headers().get("X-Ingestion-Request-RateLimit-Limit");
+        return limit == null || limit.size() == 0 ? DataSiftClient.DEFAULT_NUM : Integer.parseInt(limit.get(0));
+    }
+
+    @Override
+    public int requestRateLimitRemaining() {
+        List<String> limit = response.headers().get("X-Ingestion-Request-RateLimit-Remaining");
+        return limit == null || limit.size() == 0 ? DataSiftClient.DEFAULT_NUM : Integer.parseInt(limit.get(0));
+    }
+
+    @Override
```

Comment:
This is correct. At first I thought it would be useful to separate these two result types out to avoid exposing the rateLimitCost member & to prefix 'request' to the standard request limit members to be explicit about the type of limit being accessed (as opposed to the data limits). I can revert this if you prefer?